File: replace_fonts.js
const fs = require('fs');
const path = require('path');

const viewsDir = 'c:\\wamp64\\www\\viennabyTNQ\\v1\\views';

const googleFontsRegex = /<link[^>]*href=["']https:\/\/fonts\.googleapis\.com\/css2\?family=Cormorant\+Garamond[^"']*["'][^>]*>/g;
const newGoogleFontsLink = '<link href="https://fonts.googleapis.com/css2?family=Playfair+Display:wght@400;500;600;700&family=Lato:wght@300;400;700;900&display=swap" rel="stylesheet">';

const googleFontsRegex2 = /<link[^>]*href=["']https:\/\/fonts\.googleapis\.com\/css2\?family=Inter[^"']*Cormorant\+Garamond[^"']*["'][^>]*>/g;

function processDirectory(directory) {
    const files = fs.readdirSync(directory);
    for (const file of files) {
        const fullPath = path.join(directory, file);
        if (fs.statSync(fullPath).isDirectory()) {
            processDirectory(fullPath);
        } else if (file.endsWith('.php')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            const originalContent = content;

            content = content.replace(googleFontsRegex, newGoogleFontsLink);
            content = content.replace(googleFontsRegex2, newGoogleFontsLink);

            content = content.replace(/\['Inter'/g, "['Lato'");
            content = content.replace(/\["Inter"/g, '["Lato"');

            content = content.replace(/\['Cormorant Garamond'/g, "['Playfair Display'");
            content = content.replace(/\["Cormorant Garamond"/g, '["Playfair Display"');

            content = content.replace(/'Cormorant Garamond'/g, "['Playfair Display'");
            content = content.replace(/"Cormorant Garamond"/g, '["Playfair Display"');
            content = content.replace(/'Inter'/g, "'Lato'");
            content = content.replace(/"Inter"/g, '"Lato"');

            if (content !== originalContent) {
                fs.writeFileSync(fullPath, content, 'utf8');
                console.log(`Updated ${fullPath}`);
            }
        }
    }
}

try {
    processDirectory(viewsDir);
    console.log("Done.");
} catch (e) {
    console.error(e);
}

File: replace_fonts.py
import os
import re

views_dir = r"c:\wamp64\www\viennabyTNQ\v1\views"

google_fonts_regex = re.compile(r'<link[^>]*href=["\']https://fonts\.googleapis\.com/css2\?family=Cormorant\+Garamond[^"\']*["\'][^>]*>')
new_google_fonts_link = '<link href="https://fonts.googleapis.com/css2?family=Playfair+Display:wght@400;500;600;700&family=Lato:wght@300;400;700;900&display=swap" rel="stylesheet">'

google_fonts_regex_2 = re.compile(r'<link[^>]*href=["\']https://fonts\.googleapis\.com/css2\?family=Inter[^"\']*Cormorant\+Garamond[^"\']*["\'][^>]*>')

for root, dirs, files in os.walk(views_dir):
    for file in files:
        if file.endswith('.php'):
            filepath = os.path.join(root, file)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()

                original_content = content
                
                # Replace Google Fonts links
                content = google_fonts_regex.sub(new_google_fonts_link, content)
                content = google_fonts_regex_2.sub(new_google_fonts_link, content)
                
                # Replace Inter with Lato in tailwind config and css
                content = content.replace("['Inter'", "['Lato'")
                content = content.replace('["Inter"', '["Lato"')
                
                # Replace Cormorant Garamond with Playfair Display in tailwind config and css
                content = content.replace("['Cormorant Garamond'", "['Playfair Display'")
                content = content.replace('["Cormorant Garamond"', '["Playfair Display"')
                
                # Replace in inline CSS
                content = content.replace("'Cormorant Garamond'", "'Playfair Display'")
                content = content.replace('"Cormorant Garamond"', '"Playfair Display"')
                content = content.replace("'Inter'", "'Lato'")
                content = content.replace('"Inter"', '"Lato"')

                if content != original_content:
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(content)
                    print(f"Updated {filepath}")
            except Exception as e:
                print(f"Error accessing {filepath}: {e}")
